src/uct/ib/base/ib_iface.h

```diff
@@ -598,4 +602,14 @@ uct_ib_fence_info_init(uct_ib_fence_info_t* fence)
     fence->fence_beat = 0;
 }
 
+
+static UCS_F_ALWAYS_INLINE
+size_t uct_ib_gid_offset_of_roce_v2_addr(sa_family_t af)
```

Comment:
is it true for both ipv6 and ipv4?

Reply:
yes, this follows the same logic as `uct_ud_grh_get_dgid`, based on the "Annex17_RoCEv2 (A17.4.5.2)"... the difference is that `uct_ud_grh_get_dgid` overwrites the input parameter (presumably to save latency), whereas the new function locates the offset of the address - to be used for comparison.